flow/IThreadPoolTest.actor.cpp

```diff
@@ -60,6 +61,89 @@ TEST_CASE("/flow/IThreadPool/NamedThread") {
 	return Void();
 }
 
+struct ThreadSafePromiseStreamSender final : IThreadPoolReceiver {
+	ThreadSafePromiseStreamSender(
+		ThreadReturnPromiseStream<std::string>* notifications)
+		 : notifications(notifications) {}
+	void init() override {}
+
+	struct GetNameAction final : TypedAction<ThreadSafePromiseStreamSender, GetNameAction> {
+		double getTimeEstimate() const override { return 3.; }
+	};
+
+	void action(GetNameAction& a) {
+		pthread_t t = pthread_self();
+		const size_t arrayLen = 16;
+		char name[arrayLen];
+		int err = pthread_getname_np(t, name, arrayLen);
```

Comment:
I would probably have this return a static string, but I'm not going to insist.

Reply:
Kept it as to be consistent with the others, though I agree it doesn't really matter what we send via the promise stream.